src/internal/compilerSupport/v2.ts

```diff
@@ -309,22 +309,22 @@ export function callFunction(rawFnArgs: Array<any>, def: FunctionCallDef, script
             };
         });
 
-        // registering handler for particle timeout
-        peer.internals.regHandler.timeout(particle.id, () => {
-            reject(`Request timed out for ${def.functionName}`);
+        peer.internals.initiateParticle(particle, (stage) => {
+            if (def.returnType.tag === 'void' && (stage.stage === 'interpreted' || stage.stage === 'localWorkDone')) {
+                resolve(undefined);
+            }
```

Comment:
We've been discussing that the condition should be like this:
```
  if (stage.stage === "localWorkDone" || state === "sentToNetwork") {
    resolve()
  }
```

Right now, the condition doesn't make much sense, because `interpreted` always happens before `localWorkDone`.

Reply:
yeah, makes sense